editor/editor.go

```diff
@@ -229,6 +233,8 @@ func (e *CodeEditor) HasChanges() bool {
 }
 
 func (e *CodeEditor) LastKnownMTime() time.Time {
+	e.lock.Lock()
+	defer e.lock.Unlock()
```

Comment:
It seems reasonable to use a `sync.RWMutex` so that we can call `e.lock.RLock()` and `e.lock.RUnlock()` here.